### PR TITLE
Use new Microsoft.Graph API 

### PR DESCRIPTION
**Pull request by StrawnSC**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by panchagnula** [comment]
@StrawnSC as a part of this change can you remove the max core-cli version dependency as well? I am assuming this is already tested thoroughly?  GitHub actions sub-group commands & up with --repo

**Comment by StrawnSC** [comment]
@panchagnula I tested the original PR https://github.com/calvinsID/azure-cli-extensions/pull/85 against the core CLI's beta branch when I opened it. Since the new Microsoft.Graph code is probably in the dev branch by now, I'll test it again

**Comment by panchagnula** [comment]
yes please make sure we test this on this branch.

**Comment by StrawnSC** [comment]
@panchagnula there were some breaking changes since the graph client got moved to the dev branch. I'm fixing those up now

**Comment by StrawnSC** [comment]
@panchagnula finished making those changes and testing

File: src/containerapp/azext_containerapp/azext_metadata.json [hunk 1]
```diff
@@ -1,5 +1,4 @@
 {
     "azext.isPreview": true,
-    "azext.minCliCoreVersion": "2.15.0",
-    "azext.maxCliCoreVersion": "2.36.0"
+    "azext.minCliCoreVersion": "2.37.0"
```
Comment (panchagnula) [inline src/containerapp/azext_containerapp/azext_metadata.json:3]: Mark this breaking change when we take this to main.

**Review by panchagnula — APPROVED** [review]

**Pull request merged**